File: 프로그래머스/1/12940. 최대공약수와 최소공배수/최대공약수와 최소공배수.py
def gcd(n, m):
    if m == 0:
        return n
    else:
        return gcd(m, n%m)

def solution(n, m):
    return [gcd(n,m), gcd(n,m)*(n/gcd(n,m))*(m/gcd(n,m))]

File: 백준/Bronze/2609. 최대공약수와 최소공배수/최대공약수와 최소공배수.java
import java.io.*;
import java.util.*;

public class Main {

    private static int gcd(int a, int b) {
        if (b == 0) return a;
        return gcd(b, a % b);
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());
        int a = Integer.parseInt(st.nextToken());
        int b = Integer.parseInt(st.nextToken());

        int x = gcd(a, b);
        System.out.println(x);
        System.out.println((a/x) * (b/x) * x);
    }
}